src/common/i18n.cpp

```diff
@@ -60,9 +62,20 @@ std::string i18n_get_language()
   if (!e || !*e)
     e = "en";
 
+  // check valid values
   std::string language = e;
+  for (char c: language)
+    if (!strchr("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_-.@", c))
```

Comment:
```suggestion
    if (!strchr("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_-.@0123456789", c))
```
If LANG is set to something like ```xx_XX.UTF-8```, it will be considered invalid if ```8``` is not in the character set.
I'm not sure wether the other numbers are necessary or not; are there systems using UTF-16 or UTF-32 in LANG?
It would also be possible to move the check for valid characters after the ```language.substr``` instructions a few lines below, then checking for numbers, '@' , '-' and '.' would not be necessary.

Reply:
Good point :D